src/server/scripts/Spells/spell_paladin.cpp

```diff
@@ -2279,35 +2279,29 @@ class npc_pal_lights_hammer : public ScriptedAI
     }
 };
 
-// Consecration - 26573 and 205228
-// AreaTriggerID - 4488
-struct at_pal_consecration : AreaTriggerAI
+// Consecration - 26573 (retribution)
+// Consecration - 205228 (protection)
+class spell_pal_consecration : public AuraScript
 {
-    at_pal_consecration(AreaTrigger* areatrigger) : AreaTriggerAI(areatrigger)
-    {
-        timeInterval = 1000;
-    }
+    PrepareAuraScript(spell_pal_consecration);
 
-    void OnUpdate(uint32 diff) override
+    void OnTick(AuraEffect const* /*auraEff*/)
     {
-        Unit* caster = at->GetCaster();
-        if (!caster || !caster->IsPlayer())
-            return;
-
-        if (timeInterval <= diff)
+        if (Unit* caster = GetCaster())
         {
-            timeInterval -= diff;
-            return;
+            std::list<AreaTrigger*> ATList;
+            caster->GetAreaTriggerListWithSpellIDInRange(ATList, GetSpellInfo()->Id, caster->GetVisibilityRange());
```

Comment:
This will return all Areatriggers in visibility distance, even those created by another player

You should use std::list<AreaTrigger*> ATList = caster->GetAreaTriggers(GetSpellInfo()->Id); instead to only have the caster's ones